config/http_config.go

```diff
@@ -858,117 +854,120 @@ func (rt *basicAuthRoundTripper) CloseIdleConnections() {
 }
 
 type oauth2RoundTripper struct {
+	mtx        sync.RWMutex
+	next       *oauth2.Transport
```

Comment:
Nit (non-blocking): I'm unsure whether I like this name. It's the previous/last transport, not the next one.

Reply:
In one perspective you are right.. it's not "future". But it is a transport to call next after this round trip invocation. We use that e.g. in `NewOAuth2RoundTripper` 🙃 I think I would agree and change to `lastT` or `lastRT`? 🤔